README.md

```diff
@@ -81,30 +92,29 @@ See [Examples](#examples) for more examples.
 -   `audience`: (Optional) The value for the audience (`aud`) parameter in the
     generated GitHub Actions OIDC token. This value defaults to the value of
     `workload_identity_provider`, which is also the default value Google Cloud
-    expects for the audience parameter on the token.
+    expects for the audience parameter on the token. We do not recommend
+    changing this value.
 
--   `create_credentials_file`: (Optional) If true, the action will securely
-     generate a credentials file which can be used for authentication via gcloud
-     and Google Cloud SDKs. The default is false.
+### Authenticating via Service Account Key JSON
 
--   `activate_credentials_file`: (Optional) If true and
-    "create_credentials_file" is also true, this will set the
-    `GOOGLE_APPLICATION_CREDENTIALS` environment variable to the path to the
-    credentials file, which gcloud and Google Cloud SDKs automatically consume.
-    The default value is true.
+The following inputs are for _authenticating_ to Google Cloud via a Service
+Account Key JSON. **We recommend using Workload Identity Federation instead as
+exporting a long-lived Service Account Key JSON credential poses a security
+risk.**
 
--   `token_format`: (Optional) Output format for the generated authentication
-    token.
+-   `credentials_json`: (Required) The Google Cloud JSON service account key to use for
+    authentication. To generate access tokens or ID tokens using this service
+    account, you must grant the underlying service account
+    `roles/iam.workloadIdentityUser` permissions on itself.
```

Comment:
> you must grant the underlying service account `roles/iam.workloadIdentityUser` permissions on itself

Using `roles/iam.workloadIdentityUser` role with `credentials_json` maybe confusing to the user like we are leveraging WI here somehow. Should we just ask to grant tokenCreator instead  and maybe an example to make it clear it should be self referencing.

Reply:
`workloadIdentityUser` includes:

```
iam.serviceAccounts.get
iam.serviceAccounts.getAccessToken
iam.serviceAccounts.getOpenIdToken
iam.serviceAccounts.list
```

whereas `serviceAccountTokenCreator ` includes:

```
iam.serviceAccounts.get
iam.serviceAccounts.getAccessToken
iam.serviceAccounts.getOpenIdToken
iam.serviceAccounts.implicitDelegation
iam.serviceAccounts.list
iam.serviceAccounts.signBlob
iam.serviceAccounts.signJwt
resourcemanager.projects.get
resourcemanager.projects.list
```

I guess if we plan to support `signBlob` or `signJwt` in the future, it could be good to recommend this now. I was just concerned with granting overly broad permissions.

I'm open to either - what do you think?